docspec/src/docspec/__init__.py

```diff
@@ -207,6 +207,18 @@ class Data(ApiObject):
   #: The value of the variable as code.
   value: t.Optional[str] = None
 
+@dataclasses.dataclass
+class Indirection(Data):
+  """
+  Represents an imported name or an alias. It can be used to properly 
+  find the full name target of a link written with a local name. 
+  """
+  value: str = t.cast(str, None)
+
+  def __post_init__(self) -> None:
```

Comment:
I'm not sure if this validation is required or if `databind` with automatically catch up the different annotation and raise an error if the `value` argument is None or missing.

Reply:
Databind won't actually catch this as it's relying on the constructor of the dataclass to raise an error on missing keyword arguments. So having this validation here is good. You'd also want it for constructing the `Indirection` object manually.

Reply:
Doesn't the `__post_init__` get executed when constructing the Indirection object manually ?

Reply:
Yes it will be executed when constructing it manually and when databind constructs the object during deserialization. I was trying to say that the validation in`__post_init__` is good to keep for both cases 👍